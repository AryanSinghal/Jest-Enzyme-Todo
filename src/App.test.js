import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from './App';
import TodoInput from './components/TodoInput';

Enzyme.configure({ adapter: new Adapter() });

describe('Todo testing', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const wrapper1 = shallow(<App />);
    expect(wrapper1).toMatchSnapshot();
  });

  it('matches heading todo input', () => {
    // console.log(wrapper.debug());
    expect(wrapper.find('#heading').text()).toContain('todo input');
  });

  it('matches heading todo list', () => {
    expect(wrapper.find('#list-heading').text()).toContain('todo list');
  });

  test('Todo Input button with initial text', () => {
    expect(wrapper.find('#inputButton').text()).toContain('add item');
  });

  test('input button test', () => {
    const title = 'test';
    const button = wrapper.find('#inputButton');
    expect(button.length).toBe(1);
    const instance = wrapper.instance();
    instance.setState({ item: title });
    wrapper.find('TodoInput').prop('handleSubmit')({ preventDefault: jest.fn() });
    wrapper.update();
    const items = wrapper.state().items;
    const titles = items.map((item) => (item.title));
    const item = titles.includes(title);
    expect(item).toBe(true);
  });

  test('edit button test', () => {
    const mockState = {
      items: [{ id: '9ab55908-bf3a-4162-b7b7-cff4565bd06a', title: 'test' }],
      id: '9ab55908-bf3a-4162-b7b7-cff4565bd06a',
      item: '',
      editItem: false
    };

    const mockResult = {
      items: [],
      item: 'test',
      editItem: true,
      id: '9ab55908-bf3a-4162-b7b7-cff4565bd06a'
    }

    const instance = wrapper.instance();
    instance.setState(mockState);
    wrapper.find('TodoList').prop('handleEdit')('9ab55908-bf3a-4162-b7b7-cff4565bd06a');
    wrapper.update();
    expect(wrapper.state()).toEqual(mockResult);
  });

  test('delete button test', () => {
    const mockState = {
      items: [{ id: '9ab55908-bf3a-4162-b7b7-cff4565bd06a', title: 'test' }],
    };

    const mockResult = {
      items: [],
    }

    const instance = wrapper.instance();
    instance.setState(mockState);
    wrapper.find('TodoList').prop('handleDelete')('9ab55908-bf3a-4162-b7b7-cff4565bd06a');
    wrapper.update();
    const items = wrapper.state().items;
    expect({ items }).toEqual(mockResult);
  });

  test('TodoInput handleChange function', () => {
    const mockValue = 'test';
    const event = { target: { value: mockValue } };
    wrapper.find('TodoInput').prop('handleChange')(event);
    wrapper.update();
    const item = wrapper.state().item;
    expect(item).toBe(mockValue);
  });

  test('clearList function', () => {
    const mockData = {
      items: [
        { id: '9ab55908-bf3a-4162-b7b7-cff4565bd06a', title: 'test' },
        { id: 'e4f98052-a2db-4535-be31-de9beda51035', title: 'test2' }
      ],
      id: 'ee56337a-dbdb-4684-ad4a-8c7c8376de5c',
      item: '',
      editItem: false
    }
    const instance = wrapper.instance();
    instance.setState(mockData);
    wrapper.find('TodoList').prop('clearList')();
    wrapper.update();
    expect(wrapper.state().items).toEqual([]);
  });
});
