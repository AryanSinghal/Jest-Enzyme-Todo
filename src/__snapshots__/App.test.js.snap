// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo testing renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="col-10 mx-auto col-md-8 mt-4"
        >
          <h3
            className="text-capitalize text-center"
            id="heading"
          >
            todo input
          </h3>
          <TodoInput
            editItem={false}
            handleChange={[Function]}
            handleSubmit={[Function]}
            item=""
          />
          <TodoList
            clearList={[Function]}
            handleDelete={[Function]}
            handleEdit={[Function]}
            items={Array []}
          />
        </div>
      </div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-10 mx-auto col-md-8 mt-4"
        >
          <h3
            className="text-capitalize text-center"
            id="heading"
          >
            todo input
          </h3>
          <TodoInput
            editItem={false}
            handleChange={[Function]}
            handleSubmit={[Function]}
            item=""
          />
          <TodoList
            clearList={[Function]}
            handleDelete={[Function]}
            handleEdit={[Function]}
            items={Array []}
          />
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="text-capitalize text-center"
              id="heading"
            >
              todo input
            </h3>,
            <TodoInput
              editItem={false}
              handleChange={[Function]}
              handleSubmit={[Function]}
              item=""
            />,
            <TodoList
              clearList={[Function]}
              handleDelete={[Function]}
              handleEdit={[Function]}
              items={Array []}
            />,
          ],
          "className": "col-10 mx-auto col-md-8 mt-4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "todo input",
              "className": "text-capitalize text-center",
              "id": "heading",
            },
            "ref": null,
            "rendered": "todo input",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "editItem": false,
              "handleChange": [Function],
              "handleSubmit": [Function],
              "item": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clearList": [Function],
              "handleDelete": [Function],
              "handleEdit": [Function],
              "items": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-10 mx-auto col-md-8 mt-4"
          >
            <h3
              className="text-capitalize text-center"
              id="heading"
            >
              todo input
            </h3>
            <TodoInput
              editItem={false}
              handleChange={[Function]}
              handleSubmit={[Function]}
              item=""
            />
            <TodoList
              clearList={[Function]}
              handleDelete={[Function]}
              handleEdit={[Function]}
              items={Array []}
            />
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-10 mx-auto col-md-8 mt-4"
          >
            <h3
              className="text-capitalize text-center"
              id="heading"
            >
              todo input
            </h3>
            <TodoInput
              editItem={false}
              handleChange={[Function]}
              handleSubmit={[Function]}
              item=""
            />
            <TodoList
              clearList={[Function]}
              handleDelete={[Function]}
              handleEdit={[Function]}
              items={Array []}
            />
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="text-capitalize text-center"
                id="heading"
              >
                todo input
              </h3>,
              <TodoInput
                editItem={false}
                handleChange={[Function]}
                handleSubmit={[Function]}
                item=""
              />,
              <TodoList
                clearList={[Function]}
                handleDelete={[Function]}
                handleEdit={[Function]}
                items={Array []}
              />,
            ],
            "className": "col-10 mx-auto col-md-8 mt-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "todo input",
                "className": "text-capitalize text-center",
                "id": "heading",
              },
              "ref": null,
              "rendered": "todo input",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "editItem": false,
                "handleChange": [Function],
                "handleSubmit": [Function],
                "item": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "clearList": [Function],
                "handleDelete": [Function],
                "handleEdit": [Function],
                "items": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
